lti_consumer/lti_xblock.py

```diff
@@ -1465,7 +1481,11 @@ def _get_context_for_template(self):
         allowed_attributes = dict(bleach.sanitizer.ALLOWED_ATTRIBUTES, **{'img': ['src', 'alt']})
         sanitized_comment = bleach.clean(self.score_comment, tags=allowed_tags, attributes=allowed_attributes)
 
-        if self.lti_version == 'lti_1p1':
+        if self.lti_version == "external":
+            config = get_external_config_from_filter({}, self.external_config)
+            self.launch_url = config.get('lti_1p1_launch_url')
+
+        if self.lti_version == 'lti_1p1' or self.lti_version == "external":
             # Set launch handler depending on LTI version
             lti_block_launch_handler = self.runtime.handler_url(self, 'lti_launch_handler').rstrip('/?')
```

Comment:
Instead of running the filter here and adding a dependency to the XBlock code, instead retrieve the LTI consume and get the URL from the class:

```suggestion
        if self.lti_version == 'lti_1p1':
            consumer = self._get_lti_consumer()
            
            lti_block_launch_handler = consumer.lti_launch_url.rstrip('/?')
```
While initially, this does look more convoluted, it's simplifying the interface between the XBlock and Django app while delegating the responsibility of a consistent configuration to the `LtiConfiguration` model through `get_lti_consumer`.

Reply:
@giovannicimolin This one kind of tripped me up because there is:
1. `launch_url` - which corresponds to the URL of the tool we get from the configuration 
2. there is the the `self.runtime.handler_url` which is the XBlocks launch handler URL.

I [updated the first one](https://github.com/openedx/xblock-lti-consumer/pull/239/files#diff-cc618c2e63b8c2d293d245f9799381d9321e40fe2d05e9a4b2f80bb91995157dR1497) as per the above suggestion, the second one I have left it as it is. Fetching the XBlock's handler URL from the runtime seems appropriate for Version 1.1.